colcon_notification/event_handler/status.py

```diff
@@ -103,14 +105,19 @@ def __call__(self, event):  # noqa: D102
             line = data.line
             if isinstance(line, bytes):
                 line = line.decode(errors='replace')
-            match = self._progress_pattern.match(line)
-            if not match:
+            match_make = self._progress_pattern_make.match(line)
+            match_ninja = self._progress_pattern_ninja.match(line)
+            if not match_make and not match_ninja:
```

Comment:
I'd like to see this take a "first match wins" approach to avoid running `.match()` when we won't care.